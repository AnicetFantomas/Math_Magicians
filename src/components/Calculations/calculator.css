* {
  padding: 0;
  margin: 0;
  box-sizing: border-box;
}

section {
  background-color: rgb(18 26 31);
  background-image: url('https://www.skybounddigital.com/wp-content/uploads/2015/01/skills-background1.jpg');
  min-height: 100vh;
  width: 100%;
  display: flex;
  align-items: center;
}

.side-intro {
  margin-bottom: 35%;
  margin-left: 2%;
  color: #fff;
  font-size: 30px;
  font-weight: 700;
}

.container {
  width: 40%;
  min-width: 400px;
  background-color: rgb(39 55 59);
  border-radius: 8px;
  margin-left: 30%;
  overflow: hidden;
  font-family: 'Gill Sans', 'Gill Sans MT', Calibri, 'Trebuchet MS', sans-serif;
}

.display {
  background-color: rgb(80, 80, 80);
  height: 100px;
  width: 100%;
  text-align: right;
  color: #fff;
  padding: 20px;
  font-size: 40px;
  position: relative;
}

.all-buttons {
  color: #f5f5f5;
  display: grid;
  grid-template: repeat(4, 1fr) / repeat(4, 1fr);
}

.button {
  border: 0.5px solid rgba(92, 92, 92, 0.137);
  height: 100px;
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  font-size: 30px;
  cursor: pointer;
}

.side-col {
  background-color: orange;
}

.button:hover {
  background-color: rgb(30, 43, 46);
}

.btn-0 {
  grid-column: 1/3;
}
