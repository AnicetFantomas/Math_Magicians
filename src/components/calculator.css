* {
  padding: 0;
  margin: 0;
  box-sizing: border-box;
}

section {
  background-color: rgb(18 26 31);
  min-height: 100vh;
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
}

.container {
  width: 40%;
  min-width: 400px;
  background-color: rgb(39 55 59);
  border-radius: 8px;
  overflow: hidden;
}

.display {
  background-color: rgb(80, 80, 80);
  height: 100px;
  width: 100%;
  text-align: right;
  color: #fff;
  padding: 20px;
  font-size: 40px;
  position: relative;
}

.all-buttons {
  color: #f5f5f5;
  display: grid;
  grid-template: repeat(4, 1fr) / repeat(4, 1fr);
}

.button {
  border: 0.5px solid rgba(92, 92, 92, 0.137);
  height: 100px;
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  font-size: 30px;
  cursor: pointer;
}

.side-col {
  background-color: orange;
}

.button:hover {
  background-color: rgb(30, 43, 46);
}

.btn-0 {
  grid-column: 1/3;
}
