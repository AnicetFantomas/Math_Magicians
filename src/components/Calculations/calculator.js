import React, { useState } from 'react';
import './calculator.css';
import calculate from '../../logic/calculate';
import Navigation from '../Navigations/Navigation';

const Calculator = () => {
  const [initialCalc, setCalc] = useState({
    next: '',
    total: 0,
    operation: '',
  });
  const { next, total, operation } = initialCalc;

  const handleClick = (e) => {
    e.preventDefault();
    setCalc((state) => calculate({
      next: state.next,
      total: state.total,
      operation: state.operation,
    }, e.target.innerHTML));
  };

  return (
    <>
      <Navigation />
      <section>
        <div className="side-intro">Let&apos;s do some math !</div>
        <div className="container">
          <div className="display">
            <div className="display-result">
              {total}
              {' '}
              {operation}
              {' '}
              {next}
            </div>
          </div>
          <div className="all-buttons">
            <button type="button" className="button clear-all" onClick={handleClick}>AC</button>
            <button type="button" className="button last-entity-clear" onClick={handleClick}>+/-</button>
            <button type="button" className="button operation" onClick={handleClick}>%</button>
            <button type="button" className="button operation side-col" onClick={handleClick}>รท</button>

            <button type="button" className="button number" onClick={handleClick}>7</button>
            <button type="button" className="button number" onClick={handleClick}>8</button>
            <button type="button" className="button number" onClick={handleClick}>9</button>
            <button type="button" className="button operation side-col" onClick={handleClick}>x</button>

            <button type="button" className="button number" onClick={handleClick}>4</button>
            <button type="button" className="button number" onClick={handleClick}>5</button>
            <button type="button" className="button number" onClick={handleClick}>6</button>
            <button type="button" className="button operation side-col" onClick={handleClick}>-</button>

            <button type="button" className="button number" onClick={handleClick}>1</button>
            <button type="button" className="button number" onClick={handleClick}>2</button>
            <button type="button" className="button number" onClick={handleClick}>3</button>
            <button type="button" className="button operation side-col" onClick={handleClick}>+</button>

            <button type="button" className="button number btn-0" onClick={handleClick}>0</button>
            <button type="button" className="button number" onClick={handleClick}>.</button>
            <button type="button" className="button equal side-col" onClick={handleClick}>=</button>
          </div>
        </div>
      </section>
    </>
  );
};

export default Calculator;
