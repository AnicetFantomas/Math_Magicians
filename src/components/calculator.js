import React from 'react';
import './calculator.css';
// eslint-disable-next-line react/prefer-stateless-function
class Calculator extends React.Component {
  render() {
    return (
      <section>
        <div className="container">
          <div className="display">
            <div className="display-result">0</div>
          </div>
          <div className="all-buttons">
            <button type="button" className="button clear-all">AC</button>
            <button type="button" className="button last-entity-clear">+/-</button>
            <button type="button" className="button operation">%</button>
            <button type="button" className="button operation side-col">/</button>

            <button type="button" className="button number">7</button>
            <button type="button" className="button number">8</button>
            <button type="button" className="button number">9</button>
            <button type="button" className="button operation side-col">X</button>

            <button type="button" className="button number">4</button>
            <button type="button" className="button number">5</button>
            <button type="button" className="button number">6</button>
            <button type="button" className="button operation side-col">-</button>

            <button type="button" className="button number">1</button>
            <button type="button" className="button number">2</button>
            <button type="button" className="button number">3</button>
            <button type="button" className="button operation side-col">+</button>

            <button type="button" className="button number btn-0">0</button>
            <button type="button" className="button number">.</button>
            <button type="button" className="button equal side-col">=</button>
          </div>
        </div>
      </section>
    );
  }
}

export default Calculator;
