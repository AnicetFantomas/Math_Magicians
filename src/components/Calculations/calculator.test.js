import { render } from "@testing-library/react";
import renderer from "react-test-renderer"
import calculator from './calculator'

describe('Calculator page of App', () => {
    test('Build tje snapshot of the calculator component safely', () => {
        const rend = renderer.create(<calculator />);
        expect(rend.toJSON()).toMatchSnapshot();
    });
    test('Render Calculator component', () => {
        render(<calculator />)
    });
});