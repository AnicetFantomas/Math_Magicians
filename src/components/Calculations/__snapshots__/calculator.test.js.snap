// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator page of App Build tje snapshot of the calculator component safely 1`] = `<calculator />`;
